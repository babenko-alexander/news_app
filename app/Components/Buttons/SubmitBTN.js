import React, { PureComponent } from 'react';
import { TouchableOpacity, View, Text } from 'react-native';

import styles from './styles';

const SubmitBTN = ( text, onPress ) => (
  <TouchableOpacity
  style={{ borderRadius: 50 }}
    activeOpacity={0.7}
    onPress={onPress}
  >
    <View
    style={styles.btnBox}
    >
    <Text style={styles.btnText}>{text}</Text>
    </View>
  </TouchableOpacity>
);

export default SubmitBTN;
