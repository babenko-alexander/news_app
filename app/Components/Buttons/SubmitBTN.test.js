import React from 'react';
import {shallow} from 'enzyme';
import SubmitBTN from './SubmitBTN';

describe('SubmitBTN', () => {
        it('should match to snapshot', () => {
            const component = shallow(<SubmitBTN text="test text"/>)
            expect(component).toMatchSnapshot("SubmitBTN snapshot")
        });

    // describe('Interaction', () => {
    //     describe('onPressEvent', () => {
    //         it('should call onPress', () => {
    //             // Arrange
    //             const mockOnPress = jest.fn();      // 1. mock function
    //             const component = shallow(<SubmitBTN 
    //                 label= "test text" 
    //                 onPress={mockOnPress}           // 2. passing in mock function as props
    //             />);
    //             const instance = component.instance();  // 3. getting an instance of component

    //             // Act
    //             instance.onPressHandler();          // 4. manually triggering onPressHandler()

    //             // Assert
    //             expect(mockOnPress).toHaveBeenCalled();
    //             expect(mockOnPress).toHaveBeenCalledTimes(1);   // 5. checking return values
    //         });
    //     });
    // });

});