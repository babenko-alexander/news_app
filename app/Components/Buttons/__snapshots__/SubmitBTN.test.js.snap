// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitBTN should match to snapshot: SubmitBTN snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SubmitBTN
    text="test text"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.7,
      "children": <View>
        <Text>
          Object {
            "text": "test text",
          }
        </Text>
      </View>,
      "onPress": Object {},
      "style": Object {
        "borderRadius": 50,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Text>
          Object {
            "text": "test text",
          }
        </Text>,
        "style": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Object {
            "text": "test text",
          },
          "style": undefined,
        },
        "ref": null,
        "rendered": Object {
          "text": "test text",
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.7,
        "children": <View>
          <Text>
            Object {
              "text": "test text",
            }
          </Text>
        </View>,
        "onPress": Object {},
        "style": Object {
          "borderRadius": 50,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text>
            Object {
              "text": "test text",
            }
          </Text>,
          "style": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Object {
              "text": "test text",
            },
            "style": undefined,
          },
          "ref": null,
          "rendered": Object {
            "text": "test text",
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
